frontend/test/pytest/test_transform.py

```diff
@@ -0,0 +1,473 @@
+
+        def qnode_builder(device_name):
+            """Builder"""
+
+            @sum_expand
+            @qml.qnode(qml.device(device_name, wires=2, shots=None))
+            def qfunc():
+                """Example taken from PL tests"""
+                obs1 = qml.prod(qml.PauliX(0), qml.PauliX(1))
+                obs2 = qml.prod(qml.PauliX(0), qml.PauliY(1))
+                return [qml.expval(obs1), qml.expval(obs2)]
+
+            return qfunc
+
+        qnode_backend = qnode_builder(backend)
+        qnode_control = qnode_builder("default.qubit")
+        expected = jax.jit(qnode_control)()
+        observed = qjit(qnode_backend)()
+        assert np.allclose(expected, observed)
+
+
+class TestMergeRotations:
+    """Test Merge Rotations"""
+
+    @pytest.mark.parametrize(("theta_1", "theta_2"), [(0.3, -0.2)])
+    def test_merge_rotations(self, backend, theta_1, theta_2):
+        """Merge rotations"""
+
+        def qnode_builder(device_name):
+            """Builder"""
+
+            @qml.qnode(qml.device(device_name, wires=3))
+            @merge_rotations
+            def qfunc(theta_1, theta_2):
+                qml.RZ(theta_1, wires=0)
+                qml.RZ(theta_2, wires=0)
+                return qml.state()
+
+            return qfunc
+
+        qnode_backend = qnode_builder(backend)
+        qnode_control = qnode_builder("default.qubit")
+        expected = jax.jit(qnode_control)(theta_1, theta_2)
+        compiled_function = qjit(qnode_backend)
+        observed = compiled_function(theta_1, theta_2)
+        assert np.allclose(expected, observed)
+
+        # Here we are asserting that there is only one RZ operation
+        assert 1 == compiled_function.mlir.count('quantum.custom "RZ"')
+
+
+class TestInvalidTransform:
+    """Test validity of transforms."""
+
+    def test_split_invalid_non_commuting(self, backend):
+        """Test split non commuting"""
+
+        def qnode_builder(device_name):
+            """Builder"""
+
+            @qml.transforms.split_non_commuting
+            @qml.qnode(qml.device(device_name, wires=6), interface="jax")
+            def qfunc():
+                """Example taken from PL tests"""
+                qml.Hadamard(1)
+                qml.Hadamard(0)
+                qml.PauliZ(0)
+                # There is a measure which is a source of uncertainty!
+                measure(0)
+                qml.Hadamard(3)
+                qml.Hadamard(5)
+                qml.T(5)
+                return (
+                    qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                    qml.expval(qml.PauliX(0)),
+                    qml.expval(qml.PauliZ(1)),
+                    qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+                    qml.expval(qml.PauliX(3)),
+                    qml.expval(qml.PauliY(5)),
+                )
+
+            return qfunc
+
+        with pytest.raises(CompileError, match="Multiple tapes are generated"):
+            qjit(qnode_builder(backend))
```

Comment:
I'm confused by the test title, does this imply `split_non_commuting` is unsupported? Or is this about testing `measure` is unsupported?

Reply:
Ok looking at the test below, and assuming I'm understanding these correctly, then `test_measure_single_tape` and `test_measure_multiple_tapes` might be clearer test names, since the tests don't appear to be about a particular transform.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/280/commits/2c2aca2bedacb7b91d294f0e646db389997251f6